services/collectd/service.go

```diff
@@ -194,7 +214,9 @@ func (s *Service) Open() error {
 	}
 	s.conn = conn
 
-	s.Logger.Info(fmt.Sprint("Listening on UDP: ", conn.LocalAddr().String()))
+	if s.Diagnostic != nil {
```

Comment:
I'm really not a fan of these nil checks, Isn't it possible that they old line could have panicked in the same way that theses are trying to guard against?